batavia/core/exceptions.js

```diff
@@ -22,16 +22,61 @@ BaseException.prototype.toString = function() {
 }
 
 BaseException.prototype.__str__ = function() {
-    if (this.msg) {
-        return this.msg
+    if (this.args !== undefined) {
+        if (this.args.length == 1) {
+            return this.args[0].toString()
+        }
+        // Multiple args. Format like a tuple.
+        let result = '(' + arg.toString()
+        for (arg in this.args.slice(1)) {
+            result += ', ' + arg.toString()
+        }
+        return result = ')'
     } else {
         return ''
     }
 }
 
 BaseException.prototype.__repr__ = function() {
-    if (this.msg) {
-        return this.name + '(' + this.msg + ',)'
+    if (this.args != undefined) {
+        let output = ''
+        let count = 0
+        
+        const parse = function(msg) {
+            if (typeof msg === "string") {
+                // Message is probably a Python string.
+                // Default behaviour: wrap in single quotes
+                // String has single quotes: wrap in double quotes
+                // String has both: wrap in single quotes & escape internal single quotes.
+                let wrap = "'"
+                if (msg.includes("'")) {
+                    if (msg.includes('"')) { // example: this.msg = '\'"'
+                        msg = msg.replace("'", "\\'")
+                    } else {
+                        wrap = '"'
+                    }
+                }
+                return wrap + msg + wrap
+            } else {
+                // Avoid circular import by calling repr directly.
+                if (msg.__repr__) {
+                    return msg.__repr__()
+                }
+                return msg.toString()
+            }
+        }
+        
+        if (this.args.length) {
+            output += parse(this.args[0])
+            if (this.args.length == 1) {
+                output += ',' // A wild comma shows up in Python 3.5 and 3.6. Removed in 3.7.
```

Comment:
You can be future-proof here and wrap this in `if (version.earlier(3.7)) {}`

Reply:
And I broke this by changing version. You'll need to merge in master and change this to `!version.as_least(3.7)`.

Reply:
.at_least was added in #802 and .earlier and .later were removed. You'll need to resolve against that before merging.

Reply:
I've done this change myself now.